### DISCUSSION TITLE
Attribute [webhooks] does not exist.

### DISCUSSION THREAD

**Discussion by karaOdin** [spatie/laravel-webhook-client#208] [Q&A]
I got this error when i upgrade my project from laravel 10 to 11 using larashift

**Comment by ariddlestone** [comment]
It's been over a year, so I'm guessing @karaOdin that you either found a solution or gave up, but I've just hit this problem and thought I'd share the details/solution to my version of the problem for anyone who needs to know.

I'm seeing the error "Attribute [webhooks] does not exist" because my package which uses `Route::webhooks()` (which is required via composer rather than being in App) is having its `ServiceProvider::boot()` run before `WebhookClientServiceProvider::boot()`. This means that the macro "webhooks" has not been created when the routing for my package tries to use it.

My solution was to modify my package's RouteServiceProvider so that it loaded its routes using `$this->app->booted()`, so that it would always be run after the boot method in all service providers had been run:

```php
<?php

namespace Modules\MyModule\Providers;

use Illuminate\Support\Facades\Route;
use Illuminate\Support\ServiceProvider;

class RouteServiceProvider extends ServiceProvider
{
    protected string $name = 'MyModule';

    public function boot(): void
    {
        $this->app->booted([$this, 'onBoot']);
    }

    public function onBoot(): void
    {
        Route::middleware('api')->prefix('api')->name('api.')->group(module_path($this->name, '/routes/api.php'));
        Route::middleware('web')->group(module_path($this->name, '/routes/web.php'));
    }
?>
```
